### DISCUSSION TITLE
Encoder velocity calculation

### DISCUSSION THREAD

**Discussion by OJthe123** [Peter-van-Tol/LiteX-CNC#49] [Q&A] [answered]
Hi.
I mentioned this in another question, but it was not the main topic.

I have to multiply the encoder velocity output pin by 8 to get the right value.
Is this a bug, or do I set something wrong?

from my hal....

```
loadrt mult2 names=spindle-rpm-scale
addf spindle-rpm-scale	servo-thread
setp [LITEXCNC](NAME).encoder.00.position-scale 10000
setp [LITEXCNC](NAME).encoder.00.x4-mode 1
setp spindle-rpm-scale.in0 8
net spindle-rpm spindle-rpm-scale.in1 [LITEXCNC](NAME).encoder.00.velocity
net spindle-vel-fb-rps         =>  spindle.0.speed-in	spindle-rpm-scale.out #[LITEXCNC](NAME).encoder.00.velocity

```
when I use the encoder.00.velocity pin directly the speed is wrong

**Comment by Peter-van-Tol** [comment]
Found the cause of the error. The velocity is averaged over 8 periods, using a circular buffer. This buffer is stuck (i.e. it does not advance to the next position), therefore 7 out of 8 positions remain 0 rpm. Upgrading this discussion to an issue.

**Comment by Peter-van-Tol** [comment]
Can you check #50 whether the speed is correct now? If it works, I'll merge it into `main`

**Reply by OJthe123** [reply]
something is wrong there...linuxcnc cannot start with this drivers.
not on my lathe PC and not on the RPi4

**Reply by Peter-van-Tol** [reply]
Then it is more then just a variable. Tonight I can check on my machine and deliver a tested version.

**Reply by Peter-van-Tol** [reply]
Problem solved 👍

**Comment by Peter-van-Tol** [comment, marked as answer]
@OJthe123 : Thanks for testing. Will merge branch #50  to `main` somewhere today.
